src/nplinker/genomics/bigscape/bigscape_loader_v2.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import annotations
+import csv
+from enum import Enum
+import sqlite3
+from os import PathLike
+from nplinker.logconfig import LogConfig
+from ..abc import GCFLoaderBase
+from ..gcf import GCF
+
+
+logger = LogConfig.getLogger(__name__)
+
+class BigscapeV2GCFLoader:
+
+    def __init__(self, db_file: str | PathLike, /) -> None:
+        """Build a loader for BiG-SCAPE v2 database file.
+
+        Args:
+            cluster_file: Path to the BiG-SCAPE v2database file,
+                the filename has a pattern of "data_sqlite.db".
+
+        Attributes:
+            cluster_file: path to the BiG-SCAPE database file.
+        """
+        self.db_file = str(db_file)
+        self._gcf_list = self._parse_gcf(self.db_file)
+
+    def get_gcfs(self, keep_mibig_only: bool = False, keep_singleton: bool = False) -> list[GCF]:
+        """Get all GCF objects.
+
+        Args:
+            keep_mibig_only: True to keep GCFs that contain only MIBiG
+                BGCs.
+            keep_singleton: True to keep singleton GCFs. A singleton GCF
+                is a GCF that contains only one BGC.
+
+        Returns:
+            list[GCF]: a list of GCF objects.
```

Comment:
```suggestion
            A list of GCF objects.
```

Reply:
The static typing is not needed in docstring and the [API doc](https://nplinker.github.io/nplinker/latest/api/bigscape/) will add it automatically.